From f09fcee6b196227b1be8bd68d42b8bf76535eeda Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 22 Nov 2013 10:10:21 -0800
Subject: [PATCH 198/793] drm: xilinx: Merge 3.12 and fix conflicts

Merge 3.12 and fix conflicts. Fix the prototype of
xilinx_drm_crtc_page_flip(), remove deprecated function call(drm_fasync),
change drm_gem_cma_dumb_destroy() with drm_gem_dumb_destroy(), and
remove #include <linux/of_i2c.h>.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_crtc.c    |    3 ++-
 drivers/gpu/drm/xilinx/xilinx_drm_drv.c     |    3 +--
 drivers/gpu/drm/xilinx/xilinx_drm_encoder.c |    1 -
 3 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c b/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
index ae86388..1b6c25f 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
@@ -281,7 +281,8 @@ static void xilinx_drm_crtc_finish_page_flip(struct drm_crtc *base_crtc)
 /* page flip functions */
 static int xilinx_drm_crtc_page_flip(struct drm_crtc *base_crtc,
 				     struct drm_framebuffer *fb,
-				     struct drm_pending_vblank_event *event)
+				     struct drm_pending_vblank_event *event,
+				     uint32_t page_flip_flags)
 {
 	struct xilinx_drm_crtc *crtc = to_xilinx_crtc(base_crtc);
 	struct drm_device *drm = base_crtc->dev;
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
index ce868cc..d528004 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
@@ -299,7 +299,6 @@ static const struct file_operations xilinx_drm_fops = {
 	.unlocked_ioctl	= drm_ioctl,
 	.mmap		= drm_gem_cma_mmap,
 	.poll		= drm_poll,
-	.fasync		= drm_fasync,
 	.read		= drm_read,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl	= drm_compat_ioctl,
@@ -332,7 +331,7 @@ static struct drm_driver xilinx_drm_driver = {
 	.gem_vm_ops			= &drm_gem_cma_vm_ops,
 	.dumb_create			= drm_gem_cma_dumb_create,
 	.dumb_map_offset		= drm_gem_cma_dumb_map_offset,
-	.dumb_destroy			= drm_gem_cma_dumb_destroy,
+	.dumb_destroy			= drm_gem_dumb_destroy,
 
 	.fops				= &xilinx_drm_fops,
 
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c b/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c
index 3ea2b4a..67aabd8 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c
@@ -25,7 +25,6 @@
 #include <linux/err.h>
 #include <linux/i2c.h>
 #include <linux/of.h>
-#include <linux/of_i2c.h>
 #include <linux/platform_device.h>
 
 #include "xilinx_drm_drv.h"
-- 
1.7.5.4

